### PR TITLE
Add settings/vscode integration for diagnostic "global-element"

### PR DESCRIPTION
**Pull request by AndreaWalchshoferSCCH**
Affected settings:

- diagnostics.disable (will disable global-element as well)
- diagnostics.neededFileStatus (global-element is now an option)
- diagnostics.severity (global-element is now an option)
- diagnostics.groupfilestatus (conventions is now an option)
- diagnostics.groupseverity (conventions is now an option)

Messages are added in english language only

### CODE REVIEW

**Comment by AlWoSp** [comment]
We should add documentation at least in English in _doc/en-us/config.md_.

File: setting/schema.json [hunk 1]
```diff
@@ -395,6 +396,17 @@
                     ],
                     "type": "string"
                 },
+                "conventions": {
+                    "default": "None",
+                    "description": "* global-element\n* ...",
+                    "enum": [
```
Comment (AlWoSp) [inline setting/schema.json:402]: What is the meaning of these options?
Is there a documentation for our changes?

File: setting/schema.json [hunk 2]
```diff
@@ -1712,6 +1749,21 @@
                     ],
                     "type": "string"
                 },
+                "global-element": {
+                    "default": "Information",
+                    "description": "Enable avoiding all global elements.",
+                    "enum": [
```
Comment (AlWoSp) [inline setting/schema.json:1755]: Why are the options here different for above (lines 975-987)?

**Review by AlWoSp — CHANGES_REQUESTED** [review]

File: setting/schema.json (hunk 1 above)
Comment (AndreaWalchshoferSCCH) [reply setting/schema.json:402]: In general, the changes in this file tell VSCode that these settings exist and are valid for the settings.xml
The meaning of the different settings and their possible values are documented here: https://github.com/LuaLS/lua-language-server/wiki/Settings#diagnosticsgroupfilestatus

But maybe in short: "fileStatus" set to none means that the diagnostic (group) is disabled unless the setting is explicitly changed. The description for the group is the tooltip in VSCode, and here generally just lists the diagnostics that are contained in a certain group

**Review by AlWoSp — APPROVED** [review]

**Pull request merged**